### PR TITLE
Api fix

### LINKED ISSUE #115: seed default band. and leden.

### LINKED ISSUE #116: Fix token and security issues with the backend.

### PR DESCRIPTION
**Pull request by ManoloTonto1**
closes #116 
closes #115

### CODE REVIEW

**Comment by ManoloTonto1** [comment]
Still work in progress.

File: server/Controllers/programmeringen.cs [hunk 1]
```diff
@@ -37,26 +45,39 @@ public bool Exists(int id)
...
         return value == null ? NotFound() : value;
     }
     [HttpGet]
-    public async Task<ActionResult<IEnumerable<Programmering>>> GetAll()
+    public async Task<ActionResult<IEnumerable<Programmering>>> GetAll([FromHeader(Name = "Authorization")] string token)
     {
         var value = await context.Programmering.ToListAsync();
         return value == null ? NotFound() : value;
     }
-    [HttpGet("/count")]
```
Comment (GeneDrive) [inline server/Controllers/programmeringen.cs:51]: why does this also have the count route

File: server/Tools/Jwt.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.IdentityModel.Tokens;
+
```
Comment (GeneDrive) [inline server/Tools/Jwt.cs:5]: sadly not using my algorithm :...(

File: server/Controllers/reserveringen.cs [hunk 3]
```diff
@@ -34,35 +36,77 @@ public bool Exists(int id)
     }
 
     [HttpGet("{id}")]
-    public async Task<ActionResult<Reservering>> Get(int id)
+    public async Task<ActionResult<Reservering>> Get([FromHeader(Name = "Authorization")] string token, int id)
     {
         var value = await context.Reservering.FindAsync(id);
         return value == null ? NotFound() : value;
     }
+
```
Comment (GeneDrive) [inline server/Controllers/reserveringen.cs:44]: missing the get all which a user owns

File: server/Controllers/programmeringen.cs (hunk 1 above)
Comment (GeneDrive) [inline server/Controllers/programmeringen.cs:51]: missing get all by band/groep

Comment (GeneDrive) [inline server/Controllers/programmeringen.cs:51]: missing get all by band that betrokkenen/acteur is a part of

**Review by GeneDrive — CHANGES_REQUESTED** [review]
this is all I could find so hoper this can be fixed or explained if I missed something

File: server/Controllers/reserveringen.cs (hunk 3 above)
Comment (ManoloTonto1) [reply server/Controllers/reserveringen.cs:44]: What do you mean? when you get a user, you get the reserveringen too. i think just query that instead

Comment (GeneDrive) [reply server/Controllers/reserveringen.cs:44]: 👌

File: server/Tools/Jwt.cs (hunk 2 above)
Comment (ManoloTonto1) [reply server/Tools/Jwt.cs:5]: bc it does not make sense to use it, add more unnecessary complexity for this project.

**Review by GeneDrive — APPROVED** [review]

**Pull request merged**